src/txn/txn_rollback_to_stable.c

```diff
@@ -55,12 +55,22 @@ __rollback_delete_hs(WT_SESSION_IMPL *session, WT_ITEM *key, wt_timestamp_t ts)
     for (; ret == 0; ret = hs_cursor->prev(hs_cursor)) {
         /* Retrieve the time window from the history cursor. */
         __wt_hs_upd_time_window(hs_cursor, &hs_tw);
-        if (hs_tw->start_ts < ts)
+
+        /*
+         * Remove all history store versions with a stop timestamp greater than the start/stop
```

Comment:
The comment says strictly greater than, but the code says greater than or equal to.

Is it possible that a single transaction generates an entry in the data store and an entry in the history store, or will those be squashed together by reconciliation?

Reply:
The check is the opposite that it will break from performing the deletes if their stop timestamp is less than the stable timestamp. 

Yes, the same transaction id and same timestamp updates are squashed together.

Reply:
Ah - my mistake.